src/index.ts

```diff
@@ -85,37 +81,105 @@ export class WildlinkClient {
     return this.deviceId;
   }
 
-  public getDeviceUuid(): string {
-    return this.deviceUuid;
+  public getDevice(): Device {
+    return {
+      DeviceID: this.getDeviceId(),
+      DeviceToken: this.getDeviceToken(),
+      DeviceKey: this.getDeviceKey(),
+    };
   }
 
-  public getDomains(): Promise<Domain[]> {
+  private async createDevice(): Promise<void> {
+    // create or recreate device depending if deviceKey provided
+    const body = {
+      DeviceKey: this.deviceKey,
+    };
+    try {
+      const response = await request<Device>(`${API_URL_BASE}/v2/device`, {
+        method: 'POST',
+        headers: this.makeHeaders(),
+        body: JSON.stringify(body),
+      });
+
+      this.deviceToken = response.result.DeviceToken;
+      this.deviceKey = response.result.DeviceKey;
+      this.deviceId = response.result.DeviceID;
+    } catch (reason) {
+      return Promise.reject(reason);
+    }
+  }
+
+  public async getDomains(): Promise<ConceptDomain[]> {
     if (!this.isInit) {
-      throw new Error('WildlinkClient has not been initialized yet');
+      return Promise.reject(
+        ApplicationErrorResponse('WildlinkClient has not been initialized yet'),
+      );
     }
-    const domains = request<Domain[]>('/v2/concept/domain', {
-      method: 'GET',
-      headers: this.makeHeaders(),
-    });
-    return domains;
+
+    const response = await request<ConceptDomain[]>(
+      `${DATA_URL_BASE}/${this.applicationId}/concept-domain/1`,
+      { method: 'GET' },
+    );
+
+    return response.result;
   }
 
-  public generateVanity(url: string): Promise<Vanity> {
+  public async generateVanity(
+    url: string,
+    conceptDomain: ConceptDomain,
+    placementDetail?: string,
+  ): Promise<Vanity> {
     if (!this.isInit) {
-      throw new Error('WildlinkClient has not been initialized yet');
+      return Promise.reject(
+        ApplicationErrorResponse('WildlinkClient has not been initialized yet'),
+      );
     }
+
     if (!url) {
-      throw new Error('No URL provided');
+      return Promise.reject(ApplicationErrorResponse('No URL provided'));
+    }
+
+    if (url.indexOf(conceptDomain.Value) === -1) {
```

Comment:
`url.indexOf(conceptDomain.Value) < 0` is lil bit shorter, but no big difference